tests/libc_newlib/Makefile

```diff
@@ -2,6 +2,9 @@ include ../Makefile.tests_common
 
 USEMODULE += embunit
 
+# Make some different functions visible between newlib and newlib-nano
+CFLAGS += -D_DEFAULT_SOURCE=1
```

Comment:
CFLAGS in the application file are global. I guess this is fine, although I'd rather have the libc test compile the whole tree with the options that are default elsewhere.

Which begs the question - can we make `_DEFAULT_SOURCE=1` the default?

Reply:
> CFLAGS in the application file are global.

Not sure I understand this. Do you mean setting CFLAGS in the Makefile of a module leads to that flag being set globally? As I understand it setting this in the Makefile for a single compilation unit has only effect in that unit - and as such this should have the identical effect than the define in the source file before, right?!

About making `_DEFAULT_SOURCE=1` default: I can't tell what the side effects would be, so I can't argue for/against it. I simply fear this would be a lengthy process once more and this would starve #15931 even longer then?!

Reply:
> About making _DEFAULT_SOURCE=1 default: I can't tell what the side effects would be, so I can't argue for/against it. I simply fear this would be a lengthy process once more and this would starve #15931 even longer then?!

We could split out the global include then? At least the core part would get in, and locally globally including it will be easy while those things are fiddled out.

Reply:
I still don't understand...

Reply:
> I still don't understand...

My comment? It's exactly what you did in #15931 :). 

If its the global CFLAGS thing, I think what Kaspar means is that before only those specific headers or compilation units (when it was a pkg) had `_DEFAULT_SOURCE=1 ` so include whatever features that enable. With defining it in the application makefile all compilation units now include those set of features. And since this is a libc set test @kaspar030 would rather have it test with the same configuration used in the whole tree, which is why he suggests a global inclusion if it does not hurt.

Reply:
> > CFLAGS in the application file are global.
> 
> Not sure I understand this. Do you mean setting CFLAGS in the Makefile of a module leads to that flag being set globally?

No, setting it in a *module* makefile will set it for that module *and all modules included with DIRS*.
But the main application makefile is special. Anything set there will be set for *all modules* (globally).
There's no straight forward way to set options for just the application "module".